### PR TITLE
improve setuptools capabilities

### PR DESCRIPTION
**Pull request by 4383**
Hey!

### Overview
Using `setup.cfg` to configure project packaging. functionality introduced by [setuptools 30.3.0](http://setuptools.readthedocs.io/en/latest/setuptools.html#configuring-setup-using-setup-cfg-files).

### Features
- centralize `requirements*` management
- remove packaging configuration from python code
- using cfg file for handling project configuration
- reduce the number of files to maintains
- update python compatibilities remove python 3.2 adding python 3.6

## Benefits
- centralize configuration and dependencies
- upgrade python compatibility

Have fun!

### CODE REVIEW

**Comment by 4383** [comment]
Can you review please?

File: setup.cfg [hunk 1]
```diff
@@ -1,3 +1,65 @@
...
+
+[options]
+packages = find:
+setup_requires =
+    setuptools>=30.3.0
+include_package_data = True
+
+[options.entry_points]
+console_scripts = 
+    octodns-compare = octodns.cmds.compare:main
```
Comment (kryskool) [inline setup.cfg]: Why add octodns as an extra package ?
Not necesssary in this library

**Comment by 4383** [comment]
@kryskool done!

File: setup.cfg (hunk 1 above)
Comment (4383) [reply setup.cfg]: Just mistake

**Comment by Rbeuque74** [comment]
Hello,
As we are modifying setup.cfg, should we add wheel packaging to setup.cfg ? It will be compliant with new packaging methods of Python (cf https://pythonwheels.com/ )

Let's just add ```pip install wheel``` and
```ini
[bdist_wheel]
universal = 1

[metadata]
license_file = LICENSE
```

What do you think ?

**Comment by kryskool** [comment]
@Rbeuque74 

Not necessary to add wheel package on setup.cfg, it's a developer feature

You can add it to the [requirement-dev.txt](https://github.com/ovh/python-ovh/blob/master/requirements-dev.txt)

Regards

**Comment by 4383** [comment]
Requirement added into the setup.cfg in the sub section `extras_require` for the `dev` target

**Comment by geoffreybauduin** [comment]
Hello, in order to merge your PR, we require you to sign your commit (see https://github.com/ovh/python-ovh/blob/master/CONTRIBUTING.rst#contributing-to-python-ovh)

**Comment by 4383** [comment]
@geoffreybauduin commit signed

File: .travis.yml [hunk 2]
```diff
@@ -2,13 +2,14 @@ language: python
 python:
   - "2.6"
   - "2.7"
-  - "3.2"
   - "3.3"
   - "3.4"
   - "3.5"
+  - "3.6"
```
Comment (kryskool) [inline .travis.yml]: Maybe add nigthly version (and allow failure) like:
```
    - 'nightly'
jobs:
  allow_failures:
    - python: "nightly"
```

Comment (4383) [reply .travis.yml]: @kryskool done!

Comment (kryskool) [reply .travis.yml]: @4383 Thanks

**Comment by 4383** [comment]
Hi @Rbeuque74 can you review please?

**Comment by Rbeuque74** [comment]
Hello @4383,
Please check https://github.com/4383/python-ovh/pull/1, and if you are ok with it, we will release it :)

**Comment by 4383** [comment]
@Rbeuque74 `bdist_wheel` added

**Comment by Rbeuque74** [comment]
https://github.com/4383/python-ovh/pull/1 was containing three changes regarding wheel, and you only backported 1 of them :(
Two changes are: adding the bdist_wheel during the test procedure; adding the LICENSE file in the packaging so that it can be embed with the library.

I will add them on another PR then...

**Pull request merged**